staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -930,7 +943,7 @@ func (e *Store) Delete(ctx genericapirequest.Context, name string, options *meta
 	if err != nil {
 		return nil, false, err
 	}
-
+	e.QualifiedResource = getQualifiedResourceFromContext(ctx, e.QualifiedResource)
```

Comment:
modifying the store here is not good. Why do you need this, because `NewFunc()` uses it?